### PR TITLE
test: add event method/decorator tests

### PR DESCRIPTION
**Pull request by shiftinv**
## Summary

Adds unit tests for basic behavior of event methods and decorators, including the functionality added in #902.
~~I wrote this a few months ago, but somehow forgot about it. :>~~

## Checklist

- [x] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `pdm lint`
    - [x] I have type-checked the code by running `pdm pyright`
- [ ] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: tests/test_events.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+def client():
+    return disnake.Client()
+
+
+@pytest.fixture
+def bot():
+    return commands.Bot(
+        command_prefix=commands.when_mentioned,
+        command_sync_flags=commands.CommandSyncFlags.none(),
+    )
```
Comment (onerandomusername) [inline tests/test_events.py:22]: should these be part of conftest.py? Its up to you, really.

**Review by onerandomusername — CHANGES_REQUESTED** [review]
Rather than only test with message editing, could you please add some more events that are tested as well?

File: tests/test_events.py (hunk 1 above)
Comment (shiftinv) [reply tests/test_events.py:22]: That would make sense once we set up more comprehensive tests and proper state/client mocks, but for now it's simple enough to keep in this module.

**Comment by shiftinv** [comment]
> Rather than only test with message editing, could you please add some more events that are tested as well?

The point of this is to test the event listener framework, these tests could use essentially any event and it wouldn't make a difference.
Not sure what adding more events would accomplish here, since we're not testing the events themselves.

**Comment by onerandomusername** [comment]
> 

I'd rather we don't depend on one single event for these tests, and have at least two different events over the entire file.

**Comment by shiftinv** [comment]
> I'd rather we don't depend on one single event for these tests, and have at least two different events over the entire file.

https://github.com/DisnakeDev/disnake/pull/1016/commits/ccc8646998eb38b531a92322c77ea7ed0234c2a5

**Review by onerandomusername — APPROVED** [review]
lgtm

**Pull request merged**